ql/src/test/org/apache/hadoop/hive/ql/parse/type/TestTypeCheckProcFactory.java

```diff
@@ -140,4 +147,37 @@ public void testWithNonZeroFraction() throws Exception {
     }
   }
 
+  @Test
+  public void testValidateUDFOnTypeCheck() throws Exception {
```

Comment:
* this is a "@Parameterized" testclass; please don't add testcases which are not use the parameterized nature... move these to a new testclass
* it might make sense to split the testcases into separate methods

Reply:
Thank you for mentioning this, move the test to TestTypeCompareValidation.